core-ui/src/components/Predefined/Create/Secrets/CreateSecretForm.js

```diff
@@ -1,97 +1,223 @@
   onSubmit,
 }) {
   const { t } = useTranslation();
-  const notification = useNotification();
-  const postRequest = usePost();
   const [secret, setSecret] = useState(
-    existingSecret
-      ? yamlToSecret(existingSecret)
-      : createSecretTemplate(namespaceId),
+    existingSecret || createSecretTemplate(namespaceId),
   );
-  const [isEncoded, setEncoded] = useState(!!existingSecret);
+  const [valuesEncoded, setValuesEncoded] = useState(false);
+  const [decodeErrors, setDecodeErrors] = useState({});
+  const [lockedKeys, setLockedKeys] = useState([]);
+
   const microfrontendContext = useMicrofrontendContext();
-  const { features } = microfrontendContext;
-  const DNSExist = features?.CUSTOM_DOMAINS?.isEnabled;
 
-  const createSecret = async () => {
-    try {
-      await postRequest(
-        `/api/v1/namespaces/${namespaceId}/secrets/`,
-        secretToYaml({ secret, isEncoded }),
-      );
-      notification.notifySuccess({
-        content: t('secrets.create-modal.messages.success'),
-      });
-      LuigiClient.linkManager()
-        .fromContext('namespace')
-        .navigate(`/secrets/details/${secret.name}`);
-    } catch (e) {
-      console.error(e);
-      notification.notifyError({
-        content: t('secrets.create-modal.messages.failure', {
-          error: e.message,
-        }),
-      });
-      return false;
+  const secretDefs = getSecretDefs(t, microfrontendContext);
+  const type = secret?.type;
+  const currentDef =
+    type === 'Opaque' ? {} : secretDefs.find(def => def.type === type);
+  const secretTypes = Array.from(
+    new Set(secretDefs.map(secret => secret.type)),
+  );
+
+  useEffect(() => {
+    setLockedKeys(currentDef?.data || []);
+    setSecret({
+      ...secret,
+      data: {
+        ...(currentDef?.data || []).reduce(
+          (acc, key) => ({ ...acc, [key]: '' }),
+          {},
+        ),
+        ...(secret.data || {}),
+      },
+    });
+  }, [type]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  const toggleDecodeValues = () => {
+    setDecodeErrors({});
+    setValuesEncoded(!valuesEncoded);
+  };
+
+  const dataValue = value => {
+    if (valuesEncoded) {
+      return value?.val;
+    } else {
+      try {
+        return base64Decode(value?.val || '');
+      } catch (e) {
+        decodeErrors[value.key] = e.message;
+        setDecodeErrors(decodeErrors);
+        setValuesEncoded(true);
+        return '';
+      }
     }
   };
 
   return (
-    <CreateForm
-      title={t('secrets.create-modal.title')}
-      editMode={!!existingSecret}
-      simpleForm={
-        <SimpleForm
-          editMode={!!existingSecret}
-          secret={secret}
-          setSecret={setSecret}
-        />
-      }
-      advancedForm={
-        <AdvancedForm
-          editMode={!!existingSecret}
-          secret={secret}
-          setSecret={setSecret}
-          isEncoded={isEncoded}
-          setEncoded={setEncoded}
-        />
-      }
+    <ResourceForm
```

Comment:
Missing `createUrl={`/api/v1/namespaces/${namespaceId}/secrets/`}`, and probably `formElementRef` and `onChange`